### DISCUSSION TITLE
updating tags on existing files

### DISCUSSION THREAD

**Discussion by hmoffatt** [gerbera/gerbera#2536] [General]
I have my master music collection on my desktop, and rsync it over to my server where I run gerbera. If I make some fixes to the tags I have some trouble getting these picked up gebera.

rsync copies the new files to temporary names and then renames them over the originals. I have gerbera watching the directory with inotify; it sees the temporary rsync files and complains but that's harmless. However it doesn't seem to notice the tag changes in the files. Should it pick up changes to existing files?

The only way I've been able to get the updates to work is to remove them from the database (ideally from the PC Directory folder), then rsync them or touch them, at which point they do get imported ok. Or stop gebera, remove the db and let it import everything again.

On the subject of deleting, it seems deleting from a container other than the PC Directory folder does not remove all references to the file - is that right? And secondly is there any particular reason why the PC Directory folders don't have the delete all button that the other containers do?

Thanks!

**Comment by KarlStraussberger** [comment]
The PC Directory container is the real "home" of the media files. So when importing each item in the imported directory is mirrored to PC directory and afterwards the virtual view is created: Either buildin or based on a javascript importer. That's why you can delete a virtual item but not the real thing in PC Directory which is deleted if you delete the file from disk.

When an update occurs on the file inotify is triggered and we compare the timestamp and update the whole thing if it is newer. So if the update occurs and without change in timestamp gerbera ignores it. Otherwise we would reimport all files on restart.

The best way to get files imported again is to touch the file or rename it (or its parent folder) back and forth.

Regarding the complaints on scan and I recommend setting `ignore-extensions` https://docs.gerbera.io/en/latest/config-import.html#ignore-extensions.

**Comment by hmoffatt** [comment]
The rsync temporary files are named `.<final filename>.<random extension>`. There'd need to be extra code to optionally ignore files withe dot prefixes (sometimes described as hidden files).

So if a file is replaced by rsync, the watcher should pick it up. Do all the virtual items that refer to the old file get deleted then new ones added? I'll turn on the debug log and experiment some more. It feels like it's not quite working as I expected but I'm not certain.

**Reply by whyman** [reply]
From my perspective, this is definitely something that should work in so far as I've not noticed it not working... but I think mostly my files are "renamed" from outside a watched directory, so perhaps I am not hitting that case.

We could really do with some tests for the watcher code.
